homeassistant/components/bsblan/coordinator.py

```diff
@@ -1,57 +1,57 @@
 """DataUpdateCoordinator for the BSB-Lan integration."""
 
-from __future__ import annotations
-
 from datetime import timedelta
 from random import randint
 
 from bsblan import BSBLAN, BSBLANConnectionError
-from bsblan.models import State
 
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import CONF_HOST
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
 
 from .const import DOMAIN, LOGGER, SCAN_INTERVAL
+from .models import BSBLanCoordinatorData
 
 
-class BSBLanUpdateCoordinator(DataUpdateCoordinator[State]):
+class BSBLanUpdateCoordinator(DataUpdateCoordinator[BSBLanCoordinatorData]):
     """The BSB-Lan update coordinator."""
 
-    config_entry: ConfigEntry
-
     def __init__(
         self,
         hass: HomeAssistant,
         config_entry: ConfigEntry,
         client: BSBLAN,
     ) -> None:
         """Initialize the BSB-Lan coordinator."""
-
         self.client = client
+        self.bsblan_config_entry = config_entry
```

Comment:
Why do we do this? It's already accessible via self.config_entry. before the type was overwritten, making sure you don't have type issues